airbyte-scheduler/client/src/main/java/io/airbyte/scheduler/client/DefaultSynchronousSchedulerClient.java

```diff
@@ -102,25 +108,32 @@ public SynchronousResponse<ConnectorSpecification> createGetSpecJob(final String
         ConfigType.GET_SPEC,
         null,
         jobId -> temporalClient.submitGetSpec(UUID.randomUUID(), 0, jobSpecConfig),
+        ConnectorJobOutput::getSpec,
         null);
   }
 
   @VisibleForTesting
-  <T> SynchronousResponse<T> execute(final ConfigType configType,
-                                     @Nullable final UUID connectorDefinitionId,
-                                     final Function<UUID, TemporalResponse<T>> executor,
-                                     final UUID workspaceId) {
+  <T, U> SynchronousResponse<T> execute(final ConfigType configType,
+                                        @Nullable final UUID connectorDefinitionId,
+                                        final Function<UUID, TemporalResponse<U>> executor,
+                                        final Function<U, T> outputMapper,
+                                        final UUID workspaceId) {
```

Comment:
This method mainly remains the same, but to keep consumers of these `SynchronousResponse`s the same, I introduced an `outputMapper` function for going from the common `ConnectorJobOutput` -> the specific response we want.